examples/realtime/nextjs-authorization-demo/README.md

```diff
@@ -0,0 +1,161 @@
+$$
+  BEGIN
+    INSERT INTO public.profiles (user_id, email) VALUES (NEW.id, NEW.email); RETURN NEW;
+  END;
+$$ language plpgsql;
+
+create or replace trigger "on_new_auth_create_profile"
+after insert on auth.users for each row
+execute function insert_user ();
+
+grant execute on function insert_user () to supabase_auth_admin;
+```
+
+### RLS Rules
+We do have a lot of rules setup so we will detail them with schema.table in the section below.
+
+The most noteworthy one will be the realtime.broadcast as it will be the one used to limit access to rooms by checking if an entry for a given room name and user id exists in the table public.rooms_users.
+
+> ⚠️ All the rules here are not the most secure, you should check your use case and refine them!
+#### public.profiles
+```sql
+create policy "authenticated users can view all profiles"
+on "public"."profiles"
+as PERMISSIVE for SELECT
+to authenticated
+using ( true );
+
+create policy "authenticated user can insert profile"
+on "public"."profiles"
+as PERMISSIVE for INSERT
+to supabase_auth_admin
+with check ( true );
+
+```
+#### public.rooms_users
+```sql
+create policy "authenticated users can read rooms_users"
+on "public"."rooms_users"
+as PERMISSIVE for SELECT
+to authenticated
+using ( true );
+
+create policy "authenticated can add rooms_users"
+on "public"."rooms_users"
+as PERMISSIVE for INSERT
+to authenticated
+with check ( true );
+```
+#### realtime.channels
+```sql
+create policy "authenticated users can see all channels"
+on "realtime"."channels"
+as PERMISSIVE for SELECT
+to authenticated
+using ( true );
+
+create policy "authenticated can create channel"
+on "realtime"."channels"
+as PERMISSIVE for INSERT
+to authenticated
+with check ( true );
+```
+
+#### realtime.presence
+```sql
+create policy "authenticated users can see all presences"
+on "realtime"."presences"
+as PERMISSIVE for SELECT
+to authenticated
+using ( true );
+
+create policy "authenticated can track presences"
+on "realtime"."presences"
+as PERMISSIVE for UPDATE
+to authenticated
+using ( true );
+```
+#### realtime.broadcasts
+```sql
+create policy "authenticated user in room can access"
+on "realtime"."broadcasts"
+as PERMISSIVE for SELECT
+to authenticated
+using (
+  EXISTS (
+    SELECT ru.user_id
+    FROM rooms_users ru
+    WHERE ru.user_id = auth.uid()
+    AND ru.name = realtime.channel_name()
+  )
+);
+
+create policy "authenticated user in room can broadcast"
+on "realtime"."broadcasts"
+as PERMISSIVE for UPDATE
+to authenticated
+using (
+  EXISTS (
+    SELECT ru.user_id
+    FROM rooms_users ru
+    WHERE ru.user_id = auth.uid()
+    AND ru.name = realtime.channel_name()
+  )
+);
+```
+
+## Coding concerns
+### Need to create channels
+At the moment if you are using Realtime channels you would just connect and go. If you want your channel to respect the RLS rules setup above you will need to preemptively create the channel that will be used before connecting.
```

Comment:
We should specifically mention Broadcast and Presence and not just channel b/c a channel can use Postgres Changes and that can respect RLS rules.

Referencing "you" here is confusing b/c who's supposed to create it? The developer? Their users? I think in the demo it's the `authenticated` user.

Reply:
this can actually impact postres_changes and in theory we could even improve a bit the performance of postgres_changes if we tell users they can run some of those queries on connect 👀 

Added a warning to mention it